core/src/saros/git/JGitFacade.java

```diff
@@ -0,0 +1,192 @@
+    NullProgressMonitor monitor = NullProgressMonitor.INSTANCE;
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+    bundlewriter.writeBundle(monitor, byteArrayOutputStream);
+
+    return byteArrayOutputStream.toByteArray();
+  }
+
+  /**
+   * Fetching from a bundle to a Git repo
+   *
+   * @param bundle See <a href="https://git-scm.com/docs/git-bundle">git bundle</a>
+   * @throws IllegalArgumentException {@code workDirTree} or {@code bundle} is null and can't be
+   *     resolved
+   * @throws IOException failed while read from {@code workDirTree}, {@code bundle} handling or
+   *     while fetch
+   * @throws URISyntaxException
+   */
+  public void fetchFromBundle(byte[] bundle) throws IOException, URISyntaxException {
+    if (bundle == null) throw new IllegalArgumentException("bundle is null and can't be resolved");
+
+    try {
+      RemoteRemoveCommand remoteRemoveCommand = git.remoteRemove();
+      remoteRemoveCommand.setName("bundle");
+      remoteRemoveCommand.call();
+    } catch (GitAPIException e) {
+      throw new IOException("failed to remove old bundle path", e);
+    }
+
+    final URIish uri = new URIish("in-memory://");
+    final ByteArrayInputStream in = new ByteArrayInputStream(bundle);
+    final RefSpec rs = new RefSpec("refs/heads/*:refs/heads/*");
+    final Set<RefSpec> refs = Collections.singleton(rs);
+
+    try (TransportBundleStream transport =
+        new TransportBundleStream(git.getRepository(), uri, in)) {
+      transport.fetch(NullProgressMonitor.INSTANCE, refs);
+    } catch (org.eclipse.jgit.errors.TransportException e) {
+      throw new IOException("fetch failed", e);
+    }
+  }
+
+  /**
+   * @param revString See <a href="https://www.git-scm.com/docs/gitrevisions">gitrevisions</a>
+   * @throws IllegalArgumentException {@code workDirTree} is null can't be resolved
+   * @throws IOException failed while read
+   */
+  public String getSHA1HashByRevisionString(String revString) throws IOException {
+    return git.getRepository().resolve(revString).name();
+  }
+
+  /**
+   * Allowing to merge commits if fast forward merge is possible. The current commit has to be a
+   * ancestor of the named commit (which is accessed by resolving the given Revision String).
+   *
+   * @param revString See <a href="https://www.git-scm.com/docs/gitrevisions">gitrevisions</a>
+   * @throws IOException failed while merge
+   */
+  public void fastForwardMerge(String revString) throws IOException {
+    try {
+      git.merge()
+          .include(git.getRepository().resolve(revString))
+          .setFastForward(FastForwardMode.FF_ONLY)
+          .call();
+    } catch (IOException | RevisionSyntaxException | GitAPIException e) {
+      throw new IOException("merge failed", e);
+    }
+  }
+
+  /** @param workDirTree The directory that contains the .git directory */
+  public void setWorkDirTree(File workDirTree) {
```

Comment:
Do you have to call `setGit` after `setWorkDirTree`? Otherwise the internal workDirTree keeps the same.

Reply:
done